clojure-mode.el

```diff
@@ -1304,7 +1304,7 @@ symbol properties."
                  'clojure-indent-function)
             (get (intern-soft (match-string 1 function-name))
                  'clojure-backtracking-indent)))
-      (when (string-match (rx (or "let" "when" "while") (syntax symbol))
+      (when (string-match (rx string-start (or "let" "when" "while") (syntax symbol))
```

Comment:
I guess it's also worth adding some explanations in the code here, as to what it's aiming to achieve. You should also check if this won't break `if-let`.

Reply:
Are the new explanations something you had in mind?

I also added tests for if-let case and fixed it.